src/Snak/PropertyValueSnak.php

```diff
@@ -64,8 +64,8 @@ public function serialize() {
 	 * @param string $serialized
 	 */
 	public function unserialize( $serialized ) {
-		list( $numericId, $dataValue ) = unserialize( $serialized );
-		$this->__construct( $numericId, $dataValue );
+		list( $propertyId, $dataValue ) = unserialize( $serialized );
+		$this->__construct( self::newPropertyId( $propertyId ), $dataValue );
```

Comment:
Mark the date in your calendar. ;-) This might be the first time I don't like the fact that this relies on a protected method from a base class. Can be much simpler:
```php
$this->__construct( is_int( $propertyId ) ? $propertyId : new PropertyId( $propertyId ), $dataValue );
```